src/main/java/org/spongepowered/common/mixin/api/mcp/world/raid/RaidMixin_API.java

```diff
@@ -0,0 +1,83 @@
+package org.spongepowered.common.mixin.api.mcp.world.raid;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import net.minecraft.entity.monster.AbstractRaiderEntity;
+import net.minecraft.world.ServerBossInfo;
+import net.minecraft.world.World;
+import net.minecraft.world.raid.Raid;
+import org.spongepowered.api.boss.ServerBossBar;
+import org.spongepowered.api.data.type.RaidStatus;
+import org.spongepowered.api.raid.Wave;
+import org.spongepowered.api.world.server.ServerWorld;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Mutable;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.common.raid.SpongeWave;
+
+@Mixin(Raid.class)
+public abstract class RaidMixin_API implements org.spongepowered.api.raid.Raid {
+
+    public RaidMixin_API() {
+        for (Map.Entry<Integer, Set<AbstractRaiderEntity>> entry : this.raiders.entrySet()) {
+            api$spongeWaves.put(entry.getKey(), new SpongeWave((Raid) (Object) this, entry.getKey()));
```

Comment:
If you need to set up some fields, please do it in an `@Inject`or on the mixing, and not in the API mixins (can make a bridge method if necessary). Also, qualify with `this.`